### PR TITLE
Implemented `Store.SetLimits`

### PR DESCRIPTION
**Pull request by martindevans**
Implemented `Store.SetLimits` using the new `wasmtime_store_limiter` function.

### CODE REVIEW

**Comment by kpreisser** [comment]
Thanks for creating this PR! I also thought about implementing this, but didn't find time to do so.

> Opinions?

I would favor option 2 (use nullable values), at least for the parameters that can specify an unlimited value according to the documentation (`memory_size` and `table_elements`). This would match e.g. the `Memory.Maximum` API which is also a nullable `long?`, where `null` means the memory doesn't have a maximum.

Regarding accepting both negative values and `null`, we could throw an `ArgumentOutOfRangeException` if a negative value is specified, so that `null` is the only way to specify that the default value should be used.

What do you think?
Thanks!

File: src/Store.cs [hunk 1]
```diff
@@ -158,6 +158,49 @@ public Store(Engine engine, object? data)
...
+                throw new ArgumentOutOfRangeException(nameof(tables));
+            }
+
+            if (memories.HasValue && memories.Value < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(memories));
+            }
+
+            long tableElements64 = -1;
+            if (tableElements.HasValue)
```
Comment (peterhuene) [inline src/Store.cs:198]: Style nit: braces 😅

**Review by peterhuene — APPROVED** [review]
Looks great, just the incredibly minor nit!

File: src/Store.cs (hunk 1 above)
Comment (martindevans) [reply src/Store.cs:198]: Fixed 👍

**Comment by peterhuene** [comment]
Thanks for implementing this!

**Pull request merged**